encoder-brave/src/main/java/zipkin2/reporter/otel/brave/SpanTranslator.java

```diff
@@ -172,6 +184,28 @@ private Span.Builder builderForSingleSpan(MutableSpan span, ResourceSpans.Builde
     return spanBuilder;
   }
 
+  void resolveAddressAndPort(MutableSpan span, Supplier<String> fallbackAddress, Supplier<Integer> fallbackPort, BiConsumer<String, Integer> addressAndPortConsumer) {
+    String url = span.tag(HttpTags.URL.key());
+    String address = null;
+    int port = 0;
+    if (url != null) {
+      try {
+        URI uri = URI.create(url);
```

Comment:
style, but one way to shorten this out would be to do something like this and early return.

```
URI url = null
try {
  url  = URI.create(span.tag(HttpTags.URL.key()));
} catch (RuntimeException e) {
  return; // not present or invalid
}
```

Reply:
There are many spans that do not contain the `http.url` tag, and I think it would be expensive to generate a null pointer exception every time such a span is converted.

Reply:
good point I didn't think about that. On a similar note.. I think somewhere we have more efficient code than URI.create, just I couldn't find it quickly. Also, if we care strongly about perf probably switching to imperative than using java functional interfaces etc will reduce overhead. none of this needs to be addressed

Reply:
I used BiConsumer because I thought it might have more general-purpose uses, but in the end my only use was to set the `server.address` and `server.port`, so passing Span.Builder was enough.

Addressed in https://github.com/openzipkin-contrib/zipkin-otel/pull/28/commits/7d7449de7baa28dcdb6fb086105030138e00007e